libraries/ts-command-line/src/providers/CommandLineParser.ts

```diff
@@ -135,6 +135,9 @@ export abstract class CommandLineParser extends CommandLineParameterProvider {
    *               the process.argv will be used
    */
   public execute(args?: string[]): Promise<boolean> {
+    if (this._options.enableTabCompletionAction) {
+      this.addAction(new TabCompleteAction(this.actions, this.parameters));
```

Comment:
If `execute()` is called twice, we throw an error `'execute() was already called for this parser instance'`.  Your change here will break that, because it will first try to add the same action as second time.

Reply:
You already have `this._options.enableTabCompletionAction` in the constructor -- could you add `TabCompleteAction` in the constructor?  That would also fix the problem where `rush-lib`'s Jest snapshot doesn't include this action.  The sort order would be a bit weird, although eventually I want to alphabetize the actions (and eliminate `argparse` entirely).

Reply:
Another idea would be to change your API:  Instead of `ICommandLineParserOptions.enableTabCompletionAction`, you could make it `CommandLineParser.addTabCompletionAction()`.  This is just as concise, and it would be less magical, since the action would now get added in the exact same way as all the other actions.

Reply:
I added the TabCompleteAction instantiation in the `execute()` method so that we `this.actions` and `this.parameters` would be populated completely by all addAction() calls. I will add a flag to prevent adding the `tab-complete` action multiple times.

Reply:
The test will now call parser.execute() which ensures that `tab-completion` is added the snapshot.